test/behaviour/typeql/TypeQLSteps.java

```diff
@@ -694,34 +695,34 @@ public boolean check(Concept concept) {
             }
 
             Set<Attribute<?>> keys = concept.asThing().asRemote(tx()).getHas(true).collect(Collectors.toSet());
-            HashMap<Label, String> keyMap = new HashMap<>();
 
             for (Attribute<?> key : keys) {
-                String keyValue;
+                if (!key.getType().getLabel().equals(type))
+                    continue;
```

Comment:
I would do the filtering for this on line 697 using the Java streams API: `...getHas(true).filter(attr -> attr.getType().getLabel().equals(type)).findFirst()`. This returns an `Optional`, which if it doesn't exist you can immediately throw an useful exception